### PR TITLE
Exclude some wicked errors when reloading a connection

### PR DESCRIPTION
**Pull request by dgdavid**
Sometimes wicked returns some exit status codes that are not errors or do not mean that connection is not working. So, let's ignore them.

### CODE REVIEW

File: src/lib/wicked/adapter.js [hunk 1]
```diff
@@ -266,7 +282,17 @@ class WickedAdapter {
      * @return {Promise} Result of the operation
      */
     reloadConnection(name) {
-        return this.client.reloadConnection(name);
+        return new Promise((resolve, reject) => {
+            this.client.reloadConnection(name)
+                    .then((name) => resolve(name))
```
Comment (imobachgs) [inline src/lib/wicked/adapter.js:287]: You could simply pass the resolve function, but I guess this form is more readable.

File: src/lib/wicked/adapter.test.js [hunk 2]
```diff
@@ -119,3 +119,52 @@ describe('#interfaces', () => {
...
+                reloadConnection: reloadConnectionMock
+            };
+        });
+    });
+
+    afterAll(() => {
+        reloadConnectionMock.mockClear();
+    });
+
+    it('returns the conneciton name if everything works', async () => {
```
Comment (imobachgs) [inline src/lib/wicked/adapter.test.js:138]: typo: connection

File: src/lib/wicked/adapter.test.js [hunk 3]
```diff
@@ -119,3 +119,52 @@ describe('#interfaces', () => {
...
+
+        const client = new Client();
+        const adapter = new Adapter(client);
+
+        const name = await adapter.reloadConnection("eth0");
+        expect(name).toEqual("eth0");
+    });
+
+    it('rejects if there is a not ignored error', async () => {
+        const error = { message: "This error should be ignored", exit_status: 157 };
```
Comment (imobachgs) [inline src/lib/wicked/adapter.test.js:161]: np: "should *not* be ignored"

**Review by imobachgs — COMMENTED** [review]
Some minor comments. Otherwise, LGTM.

File: src/lib/wicked/adapter.js (hunk 1 above)
Comment (dgdavid) [reply src/lib/wicked/adapter.js:287]: > but I guess this form is more readable.

Actually not ;)

I prefer your suggestion

```js
.then(resolve)
```

than the current implementation, which is there after some changes from a _multiline resolve_ to the oneliner.

So, thanks!

**Review by imobachgs — APPROVED** [review]

**Pull request merged**